File: 23_09_2022/205._Isomorphic_Strings.py
class Solution:
    def isIsomorphic(self, s: str, t: str) -> bool:
        if(len(s)!=len(t)):
            return False
        d={}
        d1={}
        i=0
        for i in range(len(s)):
            if(s[i] not in d):
                d[s[i]]=t[i]
            if(t[i] not in d1):
                d1[t[i]]=s[i]
            if(d[s[i]]!=t[i] or d1[t[i]]!=s[i]):
                return False
            
        return True
        

File: 23_09_2022/205._Isomorphic_Strings.cpp
class Solution {
public:
    bool isIsomorphic(string s, string t) {
        
        if(s.length()!=t.length())
        {
            return false;
        }
        
        
        map<char,char> m;
        map<char,char> m1;
        
        int i=0;
        for(auto ch:s)
        {
                m[ch]=t[i++];
        }
        i=0;
        for(auto ch:t)
        {
            m1[ch]=s[i++];
        }
        
        for(i=0;i<s.length();i++)
        {
            if(m[s[i]]!=t[i] || m1[t[i]]!=s[i])
            {
                cout<<m[s[i]]<<" "<<t[i]<<" 1\n";
                cout<<m1[t[i]]<<" "<<s[i]<<" 2\n";
                return false;
            }
        }
        
        return true;
        

        
    }
};